### ISSUE TITLE
alternative storage of data

### ISSUE THREAD

**Issue by phiweger** [nanoporetech/tombo#87]
Tombo writes directly to the raw fast5. These hdf5 files are like small databases, without the protections of typical databases (regarding e.g. transactions).

While you warn users not to e.g. run multiple resquiggle instances in the same read set, I think the design decision to use the existing fast5 -- while surely efficient -- is dangerous. Using tombo now for a couple of days, I managed to successfully modify the fast5 files in a way that lets e.g. [fast5seek](https://github.com/mbhall88/fast5seek/issues/5) crash. The reasons why are hard to debug, probably some lock that is still in place or simply a corrupted fast5 read.

Would it be an idea for some future release to separate the raw fast5 data from tombo's intermediate files (I mean the information resquiggle writes to the fast5)? My first idea would be to use an sqlite database, or some database that can handle parallel writes for example. 

This is surely less efficient, but much more secure. Our current workaround is to copy the raw data, and work on the copy with tombo, which is also quite inefficient, so there is some room ;)

**Comment by marcus1487** [comment]
Yes I have certainly considered this idea. See #49 for some discussion on this topic. This will not be part of the next release, but we certainly hope to have a solution for this (separation of input and intermediate Tombo data) at some point soon.

**Comment by nhartwic** [comment]
Any progress or expected time when separation of I/O will be implemented? I don't think I can include tombo in any generalized pipelines until this issue is resolved.
